### PR TITLE
Dem solid objet and ghost cell contact search

### PR DESCRIPTION
**Pull request by OGaboriault**
# Description of the problem
- Solid object mapping criteria was to big. 
- Ghost cells weren't begin loop on in the contact_search and functions.

### CODE REVIEW

File: source/dem/dem.cc [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ DEMSolver<dim>::solve()
   // Find the smallest cell size and use this as the floating mesh mapping
   // criterion
   smallest_floating_mesh_mapping_criterion =
-    GridTools::minimal_cell_diameter(triangulation);
+    0.57735026919 * GridTools::minimal_cell_diameter(triangulation);
```
[the remark this answers is not in the document]
Comment (acdaigneault) [reply source/dem/dem.cc:1126]: In my opinion it's nice to know where the big constant came from, you can also explain why you use that it in the comment above

**Review by acdaigneault — APPROVED** [review]
Nice catch :)

File: source/dem/dem.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (blaisb) [reply source/dem/dem.cc:1126]: I fully agree with this statement :)

**Review by blaisb — APPROVED** [review]
Can you just add a comment that explains that constant and that would be chill.

**Pull request merged**